gdbplus/gdb-12.1/gdb/gdb_dep.c

```diff
@@ -48,6 +49,71 @@ inferior_memory_read (address_t addr, void* buffer, size_t sz)
 		return false;
 }
 
+void ca_switch_to_thread(struct thread_info *info)
+{
+    switch_to_thread (info);
+}
+
+int ca_num_fields(struct type *type)
+{
+    return type->num_fields();
+}
+
+const char *ca_field_name(struct type *type, int i)
+{
+    return type->field(i).name();
+}
+
+/*
+ * Get the glibc version of the debugee
+ */
+bool ca_glibc_version(int *major, int *minor)
```

Comment:
Somehow, I think this function might be better if it is moved to the ptmalloc files as it is used by ptmalloc instead of gdb (I assume `gdb_dep.c` is gdb related stuff).

Reply:
yes, it belongs to ptmalloc only. Other heaps don't care about it. I wanted to share it among multi versions of ptmalloc. We should replace it in their own common file next time.